### DISCUSSION TITLE
Cannot log out after logging in

### DISCUSSION THREAD

**Discussion by davidbrochart** [fastapi-users/fastapi-users#1059] [Q&A]
I have a strange behavior where I get a `401` when logging out, after successfully logging in. I'm using cookie authentication (with `cookie_secure=False`), and I've verified that a cookie is set at login and sent with the logout request.
Any idea what could be the issue, or how I can debug further?

**Comment by frankie567** [comment]
Let's try something. We'll check if the cookie is correctly read:

```py
current_user_token = fastapi_users.authenticator.current_user_token(active=True)

@app.get("/test-token")
async def test_token(user_token = Depends(current_user_token)):
    user, token = user_token
    return {"token": token}
```

Try to call this route with your cookie correctly set and check if it's correctly read.

**Reply by davidbrochart** [reply]
Thanks a lot, that helped me figuring out I was programmatically creating users by hand instead of following [the documentation](https://fastapi-users.github.io/fastapi-users/10.1/cookbook/create-user-programmatically/#create-a-user-programmatically), which was probably wrong.
It works fine now, although I'm still not very satisfied with the way I [update a user's password](https://github.com/jupyter-server/jupyverse/pull/206/files#diff-e4e1e78b48578ebf5900926610cc2fdf12bdfbf2dfa4578f69e46403d055b26eR89-R91). Maybe there is a better way?

**Reply by frankie567** [reply]
Unless your are manually hashing `auth_config.token`, what you are doing here basically is to store the plain-text token in the `hashed_password` column; which is clearly a security flaw. Besides, it wouldn't work if you tried to login using the token as a password since FastAPI Users would compare the hash.

Is it really important to store that token as the user password? It looks to me it's not necessary since that token comes from a global configuration and used through a completely different workflow.

I would suggest to just generate random passwords. Something like:

```py
password = user_manager.password_helper.generate()
user = user_manager.update({"password": password}, user)  # UserManager.update takes care of hashing the password
```

**Reply by davidbrochart** [reply]
You're right, it's not a very sensitive information, the Jupyter server even prints it to the terminal, so it can be stores elsewhere.
Thanks for your time!
